pithos/pithos.py

```diff
@@ -446,6 +465,32 @@ def pandora_ready(*ignore):
 
         self.worker_run('connect', args, pandora_ready, message, 'login')
 
+    def sync_explicit_content_filter_setting(self, *ignore):
+        def fn1(*ignore):
+            pass
+
+        def callback1(*ignore):
+            self.prefs_dlg.explicit_content_filter_checkbutton.set_label('Explicit Content Filter')
+            self.prefs_dlg.explicit_content_filter_checkbutton.set_sensitive(False)
+            self.prefs_dlg.explicit_content_filter_checkbutton.set_active(False)
+
+        self.worker_run(fn1, '', callback1)
+
+        if self.pandora.connected:
+            def fn2(*ignore):
+                current_filter_state = self.pandora.explicit_content_filter_state
+                self.filter_state = current_filter_state[0]
+                self.pin_protected = current_filter_state[1]
+
+            def callback2(*ignore):
+                self.prefs_dlg.explicit_content_filter_checkbutton.set_active(self.filter_state)
+                if self.pin_protected:
+                    self.prefs_dlg.explicit_content_filter_checkbutton.set_label('Explicit Content Filter - PIN Protected')
+                else:
+                    self.prefs_dlg.explicit_content_filter_checkbutton.set_sensitive(True)
+
+            self.worker_run(fn2, '', callback2)
```

Comment:
You don't need to store the return value of `fn2()` what is returned is directly passed to `callback2()` that you are ignoring. Also please pick remotely descriptive names for functions. Otherwise the usage here seems correct.